anomalib/utils/callbacks/metrics_configuration.py

```diff
@@ -23,6 +26,7 @@ class MetricsConfigurationCallback(Callback):
 
     def __init__(
         self,
+        task: str,
```

Comment:
Can we have a default value here? Otherwise we might need to propagate these changes to the benchmarking or hpo scripts as well. Or you can create a ticket and I can update those files later. Other than that the rest looks good.